docker/server/Dockerfile

```diff
@@ -14,6 +14,8 @@ RUN apt-get update \
     && apt-key adv --keyserver keyserver.ubuntu.com --recv E0C56BD4 \
     && echo $repository > /etc/apt/sources.list.d/clickhouse.list \
     && apt-get update \
+    && env DEBIAN_FRONTEND=noninteractive \
+        apt-get -y -o "Dpkg::Options::=--force-confdef" -o "Dpkg::Options::=--force-confold" upgrade \
```

Comment:
Do you have any idea why "upgrade" is needed?
Isn't it true that `apt-get install` installes the latest available version and `apt-get update` makes sure that the latest versions are available?

Reply:
If base image has some outdated system packages they will be upgraded by that command. In good scenario it will do nothing. If it will update smth it will lead to increased docker image size (it will contain not only base image + clickhouse, but also + upgradef packages).

Reply:
@filimonov Maybe simply change to 20.04?

Reply:
Yes, the idea of `upgrade` was to make sure that there are no outdated packages in the base image. I've tried to switch it to 20.04 as well. Unfortunately, all together it reduces the number of reported issues only by ~60%. One more experiment was to remove (purge) unused system packages after installing ClickHouse, but they happen to fail in their `postrm` scripts for some reason, so I didn't commit it.

So we could merge this PR because it somewhat improves the situation, but building the image from scratch as you tried in https://github.com/ClickHouse/ClickHouse/tree/master/docker/bare is probably indeed easier than fixing the minimal Ubuntu.

cc @alexey-milovidov @filimonov @alesapin